cli/config.go

```diff
@@ -202,6 +203,25 @@ func (h hypervisor) defaultVCPUs() uint32 {
 	return uint32(h.DefaultVCPUs)
 }
 
+func (h hypervisor) defaultMaxVCPUs() uint32 {
+	numcpus := goruntime.NumCPU()
+	maxvcpus := vc.MaxQemuVCPUs()
+
+	// Don't exceed the maximum number of vCPUs supported by hypervisor
+	if h.DefaultMaxVCPUs >= maxvcpus {
+		return maxvcpus
```

Comment:
what if the default is also greater than numcpus?  I think what you want is:
```
//don't exceed the number of physical CPUs. If a default is not provided, use the
// numbers of physical CPUs
if h.DefaultMaxVCPUS >= numcpus || h.DefaultMaxVCPUS == 0 {
      h.DefaultMaxVCPUs := numcpus
}

// Don't exceed the maximum number of vCPUs supported by hypervisor
if (h.DefaultMaxVCPUS > maxvcpus )  {
   return maxvcpus
}
  return h.DefaultMacVCPUS

Reply:
actually is  the same, but your code looks better, updating pr, thanks